packages/js-api/src/core/types.ts

```diff
@@ -23,6 +23,8 @@ export type StepAction =
   | 'step.start'
   | false
 
+export type PropertyPayload = Record<string | number, any>
```

Comment:
Nit: Objects can't have numeric keys, so in practice this is just `Record<string, any>`

Reply:
Great, makes sense!